File: app/src/main/java/com/google/android/leanbacklauncher/notifications/CaptivePortalNotificationCardView.java
package com.google.android.leanbacklauncher.notifications;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Bitmap.Config;
import android.graphics.Canvas;
import android.graphics.Rect;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import com.google.android.leanbacklauncher.R;
import com.google.android.tvrecommendations.TvRecommendation;

public class CaptivePortalNotificationCardView extends RecommendationView {
    private Bitmap mNetworkIcon;
    private TvRecommendation mRecommendation;

    public CaptivePortalNotificationCardView(Context context) {
        super(context);
    }

    public void bindInfoAreaTitleAndContent() {
        this.mTitleView.setText(this.mRecommendation.getTitle());
        this.mContentView.setText(this.mRecommendation.getText());
    }

    public Bitmap getContentImage() {
        if (this.mNetworkIcon == null) {
            this.mNetworkIcon = generateArtwork();
        }
        return this.mNetworkIcon;
    }

    public int getDataWidth() {
        return this.mRecommendation != null ? this.mRecommendation.getWidth() : 0;
    }

    public int getDataHeight() {
        return this.mRecommendation != null ? this.mRecommendation.getHeight() : 0;
    }

    public boolean hasProgress() {
        return false;
    }

    public void setRecommendation(TvRecommendation rec, boolean updateImage) {
        this.mRecommendation = rec;
        this.mInfoAreaColor = getResources().getColor(R.color.notif_background_color);
        bindSourceName(getResources().getString(R.string.settings_network), this.mRecommendation.getPackageName());
        bindContentDescription(this.mRecommendation.getTitle(), this.mRecommendation.getSourceName(), this.mRecommendation.getText());
        setMainImage(new BitmapDrawable(getResources(), getContentImage()));
        onBind();
    }

    public String getWallpaperUri() {
        return null;
    }

    public String getSignature() {
        if (this.mSignature == null) {
            this.mSignature = this.mRecommendation.getTitle() + this.mRecommendation.getText() + this.mRecommendation.getSourceName();
        }
        return this.mSignature;
    }

    protected void bindBadge() {
    }

    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
        boolean isLayoutRtl = true;
        super.onLayout(changed, left, top, right, bottom);
        int width = right - left;
        if (getLayoutDirection() != 1) {
            isLayoutRtl = false;
        }
        layoutMainImage(width);
        layoutSourceName(width, isLayoutRtl);
        layoutExpandedInfoArea(width, isLayoutRtl);
    }

    private Bitmap generateArtwork() {
        Drawable networkIcon = getResources().getDrawable(R.drawable.ic_settings_wifi_active_3, null);
        networkIcon.setTint(-1);
        int height = networkIcon.getIntrinsicHeight();
        int width = networkIcon.getIntrinsicWidth();
        Bitmap bmp = Bitmap.createBitmap(width, height, Config.ARGB_8888);
        bmp.eraseColor(this.mRecommendation.getColor());
        Canvas canvas = new Canvas(bmp);
        networkIcon.setBounds(new Rect(0, 0, width, height));
        networkIcon.draw(canvas);
        return bmp;
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/notifications/CaptivePortalNotificationCardView.kt
package com.amazon.tv.leanbacklauncher.notifications

import android.content.Context
import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Rect
import android.graphics.drawable.BitmapDrawable
import androidx.core.content.res.ResourcesCompat
import com.amazon.tv.leanbacklauncher.R
import com.amazon.tv.tvrecommendations.TvRecommendation
import com.bumptech.glide.request.target.SizeReadyCallback

class CaptivePortalNotificationCardView(context: Context?) : RecView(
    context!!
) {
    private var mNetworkIcon: Bitmap? = null
    private var mRecommendation: TvRecommendation? = null
    public override fun bindInfoAreaTitleAndContent() {
        mTitleView?.text = mRecommendation!!.title
        mContentView?.text = mRecommendation!!.text
    }

    override val contentImage: Bitmap?
        get() {
            if (mNetworkIcon == null) {
                mNetworkIcon = generateArtwork()
            }
            return mNetworkIcon
        }
    override val dataWidth: Int
        get() = if (mRecommendation != null) mRecommendation!!.width else 0
    override val dataHeight: Int
        get() = if (mRecommendation != null) mRecommendation!!.height else 0

    public override fun hasProgress(): Boolean {
        return false
    }

    fun setRecommendation(rec: TvRecommendation?, updateImage: Boolean) {
        mRecommendation = rec
        mInfoAreaColor = ResourcesCompat.getColor(resources, R.color.notif_background_color, null)
        mRecommendation?.let {
            bindSourceName(
                resources.getString(R.string.settings_network),
                it.packageName
            )
            bindContentDescription(
                it.title,
                it.sourceName,
                it.text
            )
        }
        setMainImage(BitmapDrawable(resources, contentImage))
        onBind()
    }

    override val wallpaperUri: String?
        get() = null

    override val signature: String?
        get() {
            if (mSignature == null) {
                mSignature =
                    mRecommendation?.title.toString() + mRecommendation?.text.toString() + mRecommendation?.sourceName.toString()
            }
            return mSignature
        }

    override fun bindBadge() {}
    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
        var isLayoutRtl = true
        super.onLayout(changed, left, top, right, bottom)
        val width = right - left
        if (layoutDirection != LAYOUT_DIRECTION_RTL) {
            isLayoutRtl = false
        }
        layoutMainImage(width)
        layoutSourceName(width, isLayoutRtl)
        layoutExpandedInfoArea(width, isLayoutRtl)
    }

    private fun generateArtwork(): Bitmap {
        val networkIcon =
            ResourcesCompat.getDrawable(resources, R.drawable.ic_settings_wifi_active_3, null)
        networkIcon!!.setTint(-1)
        val height = networkIcon.intrinsicHeight
        val width = networkIcon.intrinsicWidth
        val bmp = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
        bmp.eraseColor(mRecommendation!!.color)
        val canvas = Canvas(bmp)
        networkIcon.bounds = Rect(0, 0, width, height)
        networkIcon.draw(canvas)
        return bmp
    }

    override fun removeCallback(cb: SizeReadyCallback) {}
}